cli/args/flags.rs

```diff
@@ -2372,19 +2371,10 @@ Remote modules and multiple modules may also be specified:
 fn publish_subcommand() -> Command {
   Command::new("publish")
     .hide(true)
-    .about("Unstable preview feature: Publish a package")
+    .about("Unstable preview feature: Publish a package or workspace")
     // TODO: .long_about()
     .defer(|cmd| {
       cmd.arg(
-        Arg::new("directory")
-          .help(
-            "The directory to the package, or workspace of packages to publish",
-          )
-          .default_missing_value(".")
-          .value_hint(ValueHint::DirPath)
-          .required(true),
-      )
```

Comment:
Removed for now because it was broken. A lot of our code currently resolves stuff like the config file being resolved based on the cwd.

Reply:
After some consideration I think we really need to have this soon - maybe as a quick-fix we could spawn subprocess in the correct CWD to it working that way?

Reply:
It seems easy to just do `(cd some_dir && deno publish)`? Most people will be publishing from the directory they're in.

Reply:
I don't think I even used it in a CWD, not a big deal though. We can address it in a follow up

Reply:
If you remove this, you can’t publish single modules in a workspace anymore. That seems annoying

Reply:
You can by not publishing from the root.

Reply:
I added a test demonstrating this: https://github.com/denoland/deno/pull/21873/files#diff-4d2dc43bd5ca3db03f1a2ab469b13e12e9305f4934b714950429687eb2f09973

Reply:
@dsherret But this doesn't work if you have two deno.json (workspace level and package level), and you have import maps in both, and your package requires an import map entry from the workspace level one.

Reply:
@lucacasonato if it doesn't then that's a bug